Packs/OktaAuth0/Integrations/OktaAuth0EventCollector/OktaAuth0EventCollector.py

```diff
@@ -0,0 +1,292 @@
+            raw_response = self.get_access_token_request()
+            set_integration_context({
+                "access_token": raw_response.get("access_token"),
+                "expired_token_time": int(time.time())
+            })
+
+        return get_integration_context()["access_token"]
+
+    def get_events_request(self, query_params: dict) -> list:
+        """
+        Send request to get events from Okta Auth0.
+        """
+        access_token = self.get_access_token()
+        self._headers.update({'Authorization': f'Bearer {access_token}'})
+        raw_response = self._http_request(method='GET', url_suffix='/api/v2/logs', params=query_params)
+
+        demisto.info(f'Succesfully got {len(raw_response)} events.')
+        return raw_response
+
+    def fetch_events(self, query_params: dict, last_run: dict, fetch_events_limit: Optional[int] = 1000) -> List[dict]:
+        """
+        Aggregates logs from Okta Auth0.
+
+        Args:
+            query_params (dict): The query params to fetch the events.
+            last_run (dict): The lastRun object to update for the next fetch.
+            fetch_events_limit (int): The fetch limit parameter configured in the instance.
+
+        Return:
+            (list[dict]): The list of the aggregated events.
+        """
+        aggregated_events: List[dict] = []
+
+        events = self.get_events_request(query_params)
+        try:
+            while events:
+                for event in events:
+
+                    if len(aggregated_events) == fetch_events_limit:
+                        demisto.info(f'Reached the user-defined limit ({fetch_events_limit}) - stopping.')
+                        last_run['last_id'] = aggregated_events[-1].get('_id')
+                        break
+
+                    aggregated_events.append(event)
+
+                else:
+                    # Finished iterating through all events in this batch
+                    query_params.update({'from': aggregated_events[-1].get('_id'), 'take': 100})
+                    events = self.get_events_request(query_params)
+                    continue
+
+                demisto.info('Finished iterating through all events in this fetch run.')
+                break
+        except DemistoException as e:
+            if not e.res or e.res.status_code != 429:
+                raise e
+            demisto.info('Reached API rate limit, storing last id')
+
+        if aggregated_events:
+            last_run['last_id'] = aggregated_events[-1].get('_id')
+
+        return aggregated_events
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module_command(client: Client, params: dict) -> str:
+    """
+    Tests connection with Okta Auth0.
+    Args:
+        client (Client): The client implementing the API to Okta Auth0.
+        params (dict): The configuration parameters.
+
+    Returns:
+        (str) 'ok' if success.
+    """
+    params = prepare_query_params(params)
+    client.get_events_request(params)
```

Comment:
(I know I did it in Slack but) it's preferred to call fetch_events_command() and not only this, to check as much as possible of the code that will actually run.

Reply:
done